cachi2/core/package_managers/pip.py

```diff
@@ -851,7 +853,7 @@ def _parsed(self):
 
         return self.__parsed
 
-    def _read_lines(self):
+    def _read_lines(self) -> Generator[str, Any, None]:
```

Comment:
```suggestion
    def _read_lines(self) -> Iterator[str]:
```
https://docs.python.org/3/library/typing.html#typing.Generator

See "if you generator will only yield values...". IMO `Iterator[str]` is easier to understand than `Generator[str, None, None]`